### PR TITLE
Reorganized the documentation of the bundle

### PR DESCRIPTION
**Pull request by javiereguiluz**
We want to integrate this bundle in the symfony.com doc section (see https://github.com/symfony/symfony-docs/issues/4940). But we cannot do that with the current state of this bundle's doc. This PR tries to solve this problem.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,406 +28,5 @@ Installing this bundle can be done through these simple steps:
...
-    - connection_id - SQLite3 connection service id
-    - file_name     - SQLite3 database file name
-    - table_name    - Cache table name
-#### void
-#### xcache
-#### wincache
-#### zenddata
-
-Check the [doctrine-cache documentation Page](http://docs.doctrine-project.org/projects/doctrine-common/en/latest/reference/caching.html) for a better understanding.
+Read the [documentation](Resources/doc/index.rst) to elarn how to configure and
```
Comment (stof) [inline README.md]: typo: ``learn``

Comment (stof) [inline README.md]: and you should not link to the doc rendered on github, as you use ``.. configuration-block::``, which does not work on github (as it is a custom extension).

either don't use it here, or link to the symfony.com website

File: Resources/doc/index.rst [hunk 2]
```diff
@@ -1,88 +1,23 @@
...
-                cache_provider: metadata_cache_driver
-            query_cache_driver:
-                cache_provider: query_cache_driver
-            result_cache_driver:
-                # if you're using a version < 1.3 of the DoctrineBundle
-                # you can use the "service" type
-                type: service
-                id: doctrine_cache.providers.result_cache_driver
+    Since the version 1.3 of DoctrineBundle the DoctrineCacheBundle is included
+    by default.
```
Comment (stof) [inline Resources/doc/index.rst]: DoctrineCacheBundle is not included in DoctrineBundle. It is a dependency of DoctrineBundle. The place where it is included is in the standard edition.

And the bundle still need to be registered in the kernel if you want to use it

File: README.md (hunk 1 above)
Comment (javiereguiluz) [inline README.md]: You are right. I'll change the link once the doc is published in symfony.com.

File: Resources/doc/index.rst (hunk 2 above)
Comment (javiereguiluz) [inline Resources/doc/index.rst]: Thanks. I found that part of the original doc very confusing. It's better to remove it.

**Comment by javiereguiluz** [comment]
I find the original doc of this bundle very confusing. I propose to do some changes:

  * Merge `aliases.rst` into `usage.rst`
  * Remove `configuration.rst` because it's redundant with `usage.rst`

And I have a question: what does `acl_cache.rst` suppose to explain? I don't understand it.

**Comment by javiereguiluz** [comment]
Any thoughts about this proposal? I'd like to integrate this bundle in symfony.com doc. Thanks!

**Comment by guilhermeblanco** [comment]
@javiereguiluz is it just a reorganization of the docs? if positive, I'll merge right away... not much time free these days, sorry

**Comment by javiereguiluz** [comment]
Yes, I only moved things into separate files. I removed `configuration.rst` because its contents were repeated in other files. Maybe @stof can validate this removal?

**Comment by guilhermeblanco** [comment]
Nah... we can address later. Merging

**Pull request merged**